src/provider/view/utils/Grouping.ts

```diff
@@ -0,0 +1,101 @@
+import moment = require('moment');
+
+import {StoredNotification} from '../../model/StoredNotification';
+
+export enum GroupingType {
+    APPLICATION = 'Application',
+    DATE = 'Date'
+}
+
+/**
+ * A notification group - a set of notifications that should be grouped together
+ * within the UI.
+ *
+ * Grouping method can vary, but all grouping methods will produce a set of
+ * these groups.
+ */
+export interface Group {
+    /**
+     * Unique key for this group. Mainly used for storing a date string so as time passes we do not need to recalculate the date to group the notifications by.
+     */
+    key: string;
+    /**
+     * The grouping type used to create this group.
+     */
+    type: GroupingType;
+    /**
+     * User-visibile title that will be displayed above these notifications
+     */
+    title: string;
+
+    /**
+     * The notifications to display within this group.
+     *
+     * This array should always contain at least one notification. If there are no
+     * notifications that fit within this category then the group should be deleted.
+     */
+    notifications: StoredNotification[];
+}
+
+/**
+ * When creating a new notification group, this function determines the
+ * user-visible title for the group
+ * @param notification Notificiation
+ * @param groupingType Grouping method
+ */
+export function getGroupTitle(notification: StoredNotification, groupingType: GroupingType): string {
+    if (groupingType === GroupingType.APPLICATION) {
+        return notification.source.name || notification.source.uuid;
+    } else if (groupingType === GroupingType.DATE) {
+        const {date} = notification.notification;
+        return moment(date).calendar(undefined, {
+            sameDay: '[Today]',
+            nextDay: '[Tomorrow]',
+            nextWeek: 'dddd',
+            lastDay: '[Yesterday]',
+            lastWeek: '[Last] dddd',
+            sameElse: 'DD/MM/YYYY'
```

Comment:
Switched 'L' to which should be a localized date, I got `DD/MM/YYYY`.

Probably worth having a story for properly localizing all the dates and testing, since all the date code is pre `0.10` notifications.

Reply:
> September 13th 2019

Is being used for now.